common-libs/common/src/services/KeyManager/KeyManager.ts

```diff
@@ -0,0 +1,46 @@
+import { VCV1, VCV1SubjectBaseMA, VCV1Unsigned, VPV1, VPV1Unsigned } from '@affinidi/vc-common'
+import KeysService from "../KeysService";
+
+export type KeySuiteType = 'ecdsa' | 'rsa' | 'bbs'
+
+export type JwtObject = {
+  header: Record<string, any>
+  payload: Record<string, any>
+  signature: string
+}
+
+export type UnsignedJwtObject = Omit<JwtObject, 'signature'>
+
+export type KeyManager = {
+  signCredential<TSubject extends VCV1SubjectBaseMA>(
+    unsignedCredentialInput: VCV1Unsigned<TSubject>,
+    keySuiteType: 'ecdsa' | 'rsa' | 'bbs',
+  ): Promise<VCV1<TSubject>>
+
+  signJWTObject(unsignedJwtObject: UnsignedJwtObject, keyId?: string): Promise<JwtObject>
+
+  signPresentation(vp: VPV1Unsigned, purpose: { challenge: string; domain: string }): Promise<VPV1>
+
+  decryptByPrivateKey(encryptedMessage: string): Promise<any>
+
+  /**
+   * This method is supported only by LocalKeyManager, not recommended to use
+   * @param buffer
+   */
+  sign?(buffer: Buffer): Buffer
+
+  /**
+   * Signs a digest
+   * @param buffer
+   */
+  signAsync(buffer: Buffer): Promise<Buffer>
+
+  getAnchorTransactionPublicKey(): Promise<string>
```

Comment:
this could be sync method and implement as property `get AnchorTransactionPublicKey()`?

Reply:
I'd say no (not sure for 100%), but if this key will be given by external provider through network (e.g. external provider Aff KMS) - I think it is safe to leave async.